src/metabase/lib/cache.cljc

```diff
@@ -10,19 +10,29 @@
 
   If there is not already a key `subkey` in the map, calls `(f x)` and caches the value at `subkey`.
   If there is a value at `subkey`, it is returned directly."
-  [subkey x f]
-  (comment subkey) ; Avoids lint warning for half-unused `subkey`.
-  #?(:clj  (f x)
-     :cljs (if (or (object? x) (map? x))
-             (do
-               (when-not (.-__mbcache ^js x)
-                 (set! (.-__mbcache ^js x) (atom {})))
-               (if-let [cache (.-__mbcache ^js x)]
-                 (if-let [cached (get @cache subkey)]
-                   cached
-                   ;; Cache miss - generate the value and cache it.
-                   (let [value (f x)]
-                     (swap! cache assoc subkey value)
-                     value))
-                 (f x)))
-             (f x))))
+  ([subkey x f] (side-channel-cache subkey x f false))
+  ([subkey x f force?]
+   (comment subkey, force?) ; Avoids lint warning for half-unused inputs.
+   #?(:clj  (f x)
+      :cljs (if (or force? (object? x) (map? x))
+              (do
+                (when-not (.-__mbcache ^js x)
+                  (set! (.-__mbcache ^js x) (atom {})))
+                (when-not js/window.__mbcache_stats
+                  (set! js/window.__mbcache_stats (js-obj)))
+                (if-let [cache (.-__mbcache ^js x)]
+                  (if-let [cached (get @cache subkey)]
+                    (do
+                      (if-let [^js stats (aget js/window.__mbcache_stats subkey)]
+                        (set! (.-hits stats) (inc (.-hits stats)))
+                        (aset js/window.__mbcache_stats subkey (js-obj "hits" 1 "misses" 0)))
```

Comment:
Just curious, how is the following worse?

```suggestion
                        (aset js/window.__mbcache_stats subkey #js {:hits 1 :misses 0}))
```

Reply:
I agree it's better, actually. But this stats-tracking code got removed in any case; I don't really want to check that into master.